addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.squeezebox.handler;
+
+import java.io.Closeable;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.openhab.binding.squeezebox.internal.utils.SqueezeBoxTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/***
+ * Utility class to play a notification message. The message is added
+ * to the playlist, played and the previous state of the playlist and the
+ * player is restored.
+ *
+ * @author Mark Hilbush - Initial Contribution
+ * @author Patrik Gfeller - Utility class added reduce complexity and length of SqueezeBoxPlayerHandler.java
+ *
+ */
+class SqueezeBoxNotificationPlayer implements Closeable {
+    // An exception is thrown if we do not receive an acknowledge
+    // for a volume set command in the given amount of time [s].
+    final int volumeCommandTimeout = 4;
+
+    // We expect the media server to acknowledge a playlist command.
+    // An exception is thrown if the playlist command was not processed
+    // after the defined amount in [s]
+    final int playlistCommandTimeout = 5;
+
+    // Max length of the message in [s]. An exception is thrown if we did not
+    // receive a "stop" message from the media server.
+    final int notificationLengthTimeout = 90;
+
+    private Logger logger = LoggerFactory.getLogger(SqueezeBoxNotificationPlayer.class);
+    private SqueezeBoxPlayerState playerState;
+    private SqueezeBoxPlayerHandler squeezeBoxPlayerHandler;
+    private SqueezeBoxServerHandler squeezeBoxServerHandler;
+    private StringType uri;
+    private String mac;
+
+    boolean playlistModified;
+
+    private int notificationMessagePlaylistsIndex;
+
+    SqueezeBoxNotificationPlayer(SqueezeBoxPlayerHandler squeezeBoxPlayerHandler,
+            SqueezeBoxServerHandler squeezeBoxServerHandler, StringType uri) {
+        this.squeezeBoxPlayerHandler = squeezeBoxPlayerHandler;
```

Comment:
Every field that is set via the constructor and not overwritten somewhere else should ideally be final.